src/main/resources/Dockerfile

```diff
@@ -0,0 +1,173 @@
+##
+ # Image used to build and serve a staic documentation site.
+ # _____________________________________________________________________________
+ #
+ # Created by brightSPARK Labs
+ # www.brightsparklabs.com
+ ##
+
+# -----------------------------------------
+# BUILD STAGE: GENERATE PDF FILES
+# -----------------------------------------
+
+# This stage is not actually used in the website generation. It is just here to
+# allow PDF files to be build in a container, with Graphviz+Vega diagramming tools
+# available.
+#
+# Can be called manually via:
+#
+#   docker build -t docs -f <GENERATED_DOCKER_FILE> --target builder-pdf .
+
+FROM eclipse-temurin:17.0.9_9-jdk-jammy as builder-pdf
+
+RUN apt update
+# Allow gradle-docs plugin to read git details on files.
+RUN apt install -y git
+
+# Allow asciidoctor-diagram to render plantuml (dot) diagrams.
+RUN apt install -y graphviz
+
+# Allow asciidoctor-diagram to render Vega diagrams.
+# nvm needs an interactive shell: https://stackoverflow.com/a/60137919
+SHELL ["/bin/bash", "--login", "-i", "-c"]
+RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh | bash
+RUN source ~/.bashrc
+RUN nvm install lts/iron
+RUN npm install -g vega-cli
+
+# Get gradle distribution (done separately so Docker caches layer)
+WORKDIR /src
+COPY build.gradle settings.gradle gradlew ./
+COPY gradle ./gradle/
+RUN ./gradlew build || true # force sucess as build is expected to fail due to no sources
+
+COPY .git ./.git
+COPY src ./src
+
+# Build the PDF files.
+RUN ./gradlew bslAsciidoctorPdfVersioned --no-daemon
+
+# -----------------------------------------
+# EXPORT STAGE
+# -----------------------------------------
+
+# The image produced by this stage can be used if the files are to be exported to
+# the file system via: `docker build ... --target pdf-export-stage --output output/`
+
+# Base off scratch so output only contains the PDF files.
+FROM scratch AS pdf-export-stage
+COPY --from=builder-pdf /src/build/docs/asciidoc .
+
+# -----------------------------------------
+# BUILD STAGE: GENERATE ASCIIDOC FILES
+# -----------------------------------------
+
+FROM eclipse-temurin:17.0.9_9-jdk-alpine as builder-java
+
+# Directory where to build the image from.
+ARG BUILD_IMAGES_DIR="build/docs/images/"
```

Comment:
We can't hardcode it like this. Users can overwrite the directory in their config of the plugin.
You need to read it out of the `config` variable like we were previously doing.